rules/providers.bzl

```diff
@@ -1,6 +1,11 @@
 FrameworkInfo = provider(
     fields = {
         "vfsoverlay_infos": "Merged VFS overlay infos, present when virtual frameworks enabled",
-        "framework_headers": "Headers part of the framework's public interface",
+        "binary": "The binary of the framework",
```

Comment:
This one is returned inside of the apple binary already - any reason to replicate here?

Reply:
I want to reuse FrameworkInfo to package a dynamic framework. By adding binary and swiftmodule here, I won't need to pass around AppleBundleInfo or swiftinfo.

Reply:
What I'm saying is that today, upstream dependees of this rule are required to retrieve the swiftmodule, and binaries. 
https://github.com/bazel-ios/rules_ios/blob/master/rules/framework.bzl#L497 - so it's already passed around. To access the attributes, just use `framework[SwiftInfo].swiftmodule` or the other existing ones

Reply:
Totally understood. I want to make FrameworkInfo self-contain from the API's perspective. Then downstream targets can only return FrameworkInfo without returning SwiftInfo or AppleBundleInfo.